#VRML_SIM R2022a utf8
WorldInfo {
  info [
    "A simple RCJ Soccer game simulated using Webots."
  ]
  title "RCJ Soccer Sim"
  optimalThreadCount 8
}
Viewpoint {
  orientation -0.6784263609187928 -0.0037408675590809596 0.7346588859602727 3.149694480755937
  position 0.2996552824369915 0.025347941299924653 3.1323965904040656
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF SOCCER_FIELD Solid {
  children [
    DEF GROUND Transform {
      rotation 0 0 1 1.5707996938995747
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "soccer/soccer-256.jpg"
              ]
            }
            roughness 1
            metalness 0
          }
          geometry Plane {
            size 1.5 1.5
          }
        }
      ]
    }
    DEF YELLOW_GOAL_GROUND Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "soccer/soccer-256.jpg"
          ]
        }
        roughness 1
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.75 -0.95 0
            -0.75 -0.75 0
            0.75 -0.75 0
            0.75 -0.95 0
          ]
        }
        texCoord TextureCoordinate {
          point [
            0 0
            0 0
            0 0
            0 0
          ]
        }
        ccw FALSE
        coordIndex [
          0, 1, 2, 3
        ]
        texCoordIndex [
          0, 1, 2, 3
        ]
      }
    }
    DEF BLUE_GOAL_GROUND Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "soccer/soccer-256.jpg"
          ]
        }
        roughness 1
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.75 0.95 0
            -0.75 0.75 0
            0.75 0.75 0
            0.75 0.95 0
          ]
        }
        texCoord TextureCoordinate {
          point [
            0 0
            0 0
            0 0
            0 0
          ]
        }
        coordIndex [
          0, 1, 2, 3
        ]
        texCoordIndex [
          0, 1, 2, 3
        ]
      }
    }
    DEF NORTH_WALL Shape {
      appearance DEF WALL_APPEARANCE PBRAppearance {
        baseColor 0.388235 0.388235 0.388235
        roughness 1
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.66 -0.76 0
            -0.21 -0.76 0
            -0.21 -0.75 0
            -0.59 -0.75 0
            -0.65 -0.69 0
            -0.65 0.69 0
            -0.59 0.75 0
            -0.21 0.75 0
            -0.21 0.76 0
            -0.65 0.76 0
            -0.66 0.76 0
            -0.66 -0.76 0.09
            -0.21 -0.76 0.09
            -0.21 -0.75 0.09
            -0.59 -0.75 0.09
            -0.65 -0.69 0.09
            -0.65 0.69 0.09
            -0.59 0.75 0.09
            -0.21 0.75 0.09
            -0.21 0.76 0.09
            -0.65 0.76 0.09
            -0.66 0.76 0.09
          ]
        }
        coordIndex [
          0, 10, 9, -1, 9, 8, 7, -1, 9, 7, 6, -1, 2, 1, 0, -1, 3, 2, 0, -1, 9, 6, 5, -1, 4, 3, 0, -1, 0, 9, 5, -1, 0, 5, 4, -1, 11, 12, 13, -1, 11, 13, 14, -1, 20, 21, 11, -1, 18, 19, 20, -1, 17, 18, 20, -1, 11, 14, 15, -1, 16, 17, 20, -1, 11, 15, 16, -1, 11, 16, 20, -1, 0, 1, 12, 11, -1, 1, 2, 13, 12, -1, 2, 3, 14, 13, -1, 3, 4, 15, 14, -1, 4, 5, 16, 15, -1, 5, 6, 17, 16, -1, 6, 7, 18, 17, -1, 7, 8, 19, 18, -1, 8, 9, 20, 19, -1, 9, 10, 21, 20, -1, 10, 0, 11, 21, -1
        ]
      }
    }
    DEF SOUTH_WALL Shape {
      appearance USE WALL_APPEARANCE
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0.66 -0.76 0
            0.66 0.76 0
            0.65 0.76 0
            0.21 0.76 0
            0.21 0.75 0
            0.59 0.75 0
            0.65 0.69 0
            0.65 -0.69 0
            0.59 -0.75 0
            0.21 -0.75 0
            0.21 -0.76 0
            0.66 -0.76 0.09
            0.66 0.76 0.09
            0.65 0.76 0.09
            0.21 0.76 0.09
            0.21 0.75 0.09
            0.59 0.75 0.09
            0.65 0.69 0.09
            0.65 -0.69 0.09
            0.59 -0.75 0.09
            0.21 -0.75 0.09
            0.21 -0.76 0.09
          ]
        }
        coordIndex [
          0, 10, 9, -1, 0, 9, 8, -1, 2, 1, 0, -1, 4, 3, 2, -1, 5, 4, 2, -1, 0, 8, 7, -1, 6, 5, 2, -1, 0, 7, 6, -1, 0, 6, 2, -1, 11, 12, 13, -1, 13, 14, 15, -1, 13, 15, 16, -1, 20, 21, 11, -1, 19, 20, 11, -1, 13, 16, 17, -1, 18, 19, 11, -1, 11, 13, 17, -1, 11, 17, 18, -1, 0, 1, 12, 11, -1, 1, 2, 13, 12, -1, 2, 3, 14, 13, -1, 3, 4, 15, 14, -1, 4, 5, 16, 15, -1, 5, 6, 17, 16, -1, 6, 7, 18, 17, -1, 7, 8, 19, 18, -1, 8, 9, 20, 19, -1, 9, 10, 21, 20, -1, 10, 0, 11, 21, -1
        ]
      }
    }
    DEF YELLOW_GOAL Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.21 -0.86 0
            0.21 -0.86 0
            0.21 -0.75 0
            0.2 -0.75 0
            0.2 -0.85 0
            -0.2 -0.85 0
            -0.2 -0.75 0
            -0.21 -0.75 0
            -0.21 -0.86 0
            -0.21 -0.86 0.12
            0.21 -0.86 0.12
            0.21 -0.75 0.12
            0.2 -0.75 0.12
            0.2 -0.85 0.12
            -0.2 -0.85 0.12
            -0.2 -0.75 0.12
            -0.21 -0.75 0.12
            -0.21 -0.86 0.12
          ]
        }
        coordIndex [
          0, 1, 2, 3, 4, 5, 6, 7, 8, -1, 0, 1, 10, 9, -1, 1, 2, 11, 10, -1, 2, 3, 12, 11, -1, 3, 4, 13, 12, -1, 4, 5, 14, 13, -1, 5, 6, 15, 14, -1, 6, 7, 16, 15, -1, 7, 8, 17, 16, -1
        ]
      }
    }
    DEF BLUE_GOAL Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 1
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.21 0.86 0
            -0.21 0.75 0
            -0.2 0.75 0
            -0.2 0.85 0
            0.2 0.85 0
            0.2 0.75 0
            0.21 0.75 0
            0.21 0.86 0
            -0.21 0.86 0
            -0.21 0.86 0.12
            -0.21 0.75 0.12
            -0.2 0.75 0.12
            -0.2 0.85 0.12
            0.2 0.85 0.12
            0.2 0.75 0.12
            0.21 0.75 0.12
            0.21 0.86 0.12
            -0.21 0.86 0.12
          ]
        }
        coordIndex [
          0, 1, 2, 3, 4, 5, 6, 7, 8, -1, 0, 1, 10, 9, -1, 1, 2, 11, 10, -1, 2, 3, 12, 11, -1, 3, 4, 13, 12, -1, 4, 5, 14, 13, -1, 5, 6, 15, 14, -1, 6, 7, 16, 15, -1, 7, 8, 17, 16, -1
        ]
      }
    }
  ]
  name "soccer field"
  boundingObject Group {
    children [
      USE GROUND
      DEF NOTH_BBOX Transform {
        translation 0.655 0 0.045
        children [
          Box {
            size 0.01 1.5 0.09
          }
        ]
      }
      DEF SOUTH_BBOX Transform {
        translation -0.655 0 0.045
        children [
          Box {
            size 0.01 1.5 0.09
          }
        ]
      }
      DEF NW_BBOX Transform {
        translation 0.43 0.755 0.045
        children [
          Box {
            size 0.435 0.01 0.09
          }
        ]
      }
      DEF NE_BBOX Transform {
        translation 0.43 -0.755 0.045
        children [
          Box {
            size 0.435 0.01 0.09
          }
        ]
      }
      DEF SW_BBOX Transform {
        translation -0.43 0.755 0.045
        children [
          Box {
            size 0.435 0.01 0.09
          }
        ]
      }
      DEF SE_BBOX Transform {
        translation -0.43 -0.755 0.045
        children [
          Box {
            size 0.435 0.01 0.09
          }
        ]
      }
      DEF NWC_BBOX Transform {
        translation -0.62 -0.725 0.045
        rotation 0 0 1 -0.7853
        children [
          Box {
            size 0.09 0.01 0.09
          }
        ]
      }
      DEF NEC_BBOX Transform {
        translation -0.62 0.725 0.045
        rotation 0 0 1 0.7853
        children [
          Box {
            size 0.09 0.01 0.09
          }
        ]
      }
      DEF SWC_BBOX Transform {
        translation 0.62 -0.725 0.045
        rotation 0 0 1 0.7853
        children [
          Box {
            size 0.09 0.01 0.09
          }
        ]
      }
      DEF SEC_BBOX Transform {
        translation 0.62 0.725 0.045
        rotation 0 0 1 -0.7853
        children [
          Box {
            size 0.09 0.01 0.09
          }
        ]
      }
      DEF YG_BBOX Transform {
        translation 0 0.855 0.061
        children [
          Box {
            size 0.42 0.01 0.12
          }
        ]
      }
      DEF BG_BBOX Transform {
        translation 0 -0.855 0.061
        children [
          Box {
            size 0.42 0.01 0.12
          }
        ]
      }
      DEF BGS_BBOX Transform {
        translation -0.205 -0.8 0.061
        children [
          Box {
            size 0.01 0.1 0.12
          }
        ]
      }
      DEF BGN_BBOX Transform {
        translation 0.205 -0.8 0.061
        children [
          Box {
            size 0.01 0.1 0.12
          }
        ]
      }
      DEF YGS_BBOX Transform {
        translation -0.205 0.8 0.061
        children [
          Box {
            size 0.01 0.1 0.12
          }
        ]
      }
      DEF YGN_BBOX Transform {
        translation 0.205 0.8 0.061
        children [
          Box {
            size 0.01 0.1 0.12
          }
        ]
      }
    ]
  }
  locked TRUE
}
DEF BALL Robot {
  translation -0.167108 0.423784 -0.00363175
  rotation 0.6990446464682792 0.7139696268773023 0.03979892135126668 -2.52489534947534
  children [
    DEF FOOTBALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "soccer/football_base_color.jpg"
          ]
        }
        roughness 0.4
        metalness 0
        normalMap ImageTexture {
          url [
            "soccer/football_normal.jpg"
          ]
        }
      }
      geometry Sphere {
        radius 0.021
        subdivision 3
      }
    }
    Emitter {
      name "ball emitter"
      type "infra-red"
      range 0.7
      channel 4
      baudRate 115200
      allowedChannels [
        4
      ]
    }
    Emitter {
      name "ball emitter short"
      type "infra-red"
      range 0.13
      channel 5
      baudRate 115200
      allowedChannels [
        5
      ]
    }
  ]
  name "soccer ball"
  model "soccer ball"
  boundingObject USE FOOTBALL_SHAPE
  physics Physics {
    density -1
    mass 0.05
  }
  recognitionColors [
    1 1 1
    0 0 0
  ]
  controller "rcj_soccer_ball"
  synchronization FALSE
}
DEF Y1 Robot {
  translation -0.0221481 -0.456864 0.0348618
  rotation -0.05663093809517209 0.05691655936473967 0.9967715094847669 1.5733562792287588
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        position -2.6781024286282286e-11
        axis 0 1 0
        anchor 0 -0.045 -0.0176
      }
      device [
        RotationalMotor {
          name "left wheel motor"
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 1.1102230246251565e-16 -0.044999999999999984 -0.0176
        rotation -3.9446408034225584e-11 3.9446263139744414e-11 -1 1.5708
        children [
          Transform {
            rotation -0.5771969549958951 0.5776567755050862 -0.5771969549958951 2.0939354039397986
            children [
              DEF WHEEL_SHAPE Shape {
                appearance PBRAppearance {
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "right wheel"
        boundingObject Transform {
          rotation -0.5771969549958951 0.5776567755050862 -0.5771969549958951 2.0939354039397986
          children [
            USE WHEEL_SHAPE
          ]
        }
        physics DEF WHEEL_PHYSICS Physics {
          density -1
          mass 0.005
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -2.6972319995644694e-11
        axis 0 1 0
        anchor 0 0.045 -0.0176
      }
      device [
        RotationalMotor {
          name "right wheel motor"
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.044999999999999984 -0.0176
        rotation -3.820025743926462e-11 3.820011712214175e-11 -1 1.5708
        children [
          Transform {
            rotation -0.5771969549958951 0.5776567755050862 -0.5771969549958951 2.0939354039397986
            children [
              Shape {
                appearance PBRAppearance {
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject Transform {
          rotation -0.5771969549958951 0.5776567755050862 -0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.02
              }
            }
          ]
        }
        physics USE WHEEL_PHYSICS
      }
    }
    DEF YELLOW_ROBOT_SHAPE Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "soccer/yellow.png"
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
    Solid {
      translation 0 0.0385 0.02
      rotation 0 0 1 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
        DistanceSensor {
          name "distancesensor left"
          lookupTable [
            0 0 0
            1 1000 0.05
          ]
          type "sonar"
        }
      ]
      name "left sonar"
    }
    Solid {
      translation 0 -0.0385 0.02
      rotation 0 0 1 -1.5707996938995747
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
        DistanceSensor {
          name "distancesensor right"
          lookupTable [
            0 0 0
            1 1000 0.05
          ]
          type "sonar"
        }
      ]
      name "right sonar"
    }
    Solid {
      translation 0.0385 0 0.02
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
        DistanceSensor {
          name "distancesensor front"
          lookupTable [
            0 0 0
            1 1000 0.05
          ]
          type "sonar"
        }
      ]
      name "front sonar"
    }
    Solid {
      translation -0.0385 0 0.02
      rotation 0 0 1 3.141592653589793
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
        DistanceSensor {
          name "distancesensor back"
          lookupTable [
            0 0 0
            1 1000 0.05
          ]
          type "sonar"
        }
      ]
      name "back sonar"
    }
    Receiver {
      name "supervisor receiver"
      channel 1
      allowedChannels [
        1
      ]
    }
    Receiver {
      name "team receiver"
      channel 3
      allowedChannels [
        3
      ]
    }
    Receiver {
      name "ball receiver"
      type "infra-red"
      channel 4
      allowedChannels [
        4
      ]
    }
    Emitter {
      name "team emitter"
      range 10
      channel 3
      baudRate 115200
      allowedChannels [
        3
      ]
    }
    GPS {
    }
    Compass {
    }
  ]
  name "Y1"
  boundingObject USE YELLOW_ROBOT_SHAPE
  physics DEF ROBOT_PHYSICS Physics {
    density -1
    mass 0.2
  }
  controller "robots"
  synchronization FALSE
}
DEF Y2 Robot {
  translation -0.0074161 0.33495 0.0420318
  rotation 0.00030649807789454465 -0.0004901646006941137 0.9999998328987825 1.5700002866645308
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2.240096512680971e-10
        axis 0 1 0
        anchor 0 -0.045 -0.0176
      }
      device [
        RotationalMotor {
          name "left wheel motor"
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 5.551115123125783e-17 -0.04499999999999993 -0.0176
        rotation -1.5515474369313838e-10 1.551541737789887e-10 -1 1.5708
        children [
          Transform {
            rotation -0.5771969549958951 0.5776567755050862 -0.5771969549958951 2.0939354039397986
            children [
              DEF WHEEL_SHAPE Shape {
                appearance PBRAppearance {
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "right wheel"
        boundingObject Transform {
          rotation -0.5771969549958951 0.5776567755050862 -0.5771969549958951 2.0939354039397986
          children [
            USE WHEEL_SHAPE
          ]
        }
        physics DEF WHEEL_PHYSICS Physics {
          density -1
          mass 0.005
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -5.175981398342014e-09
        axis 0 1 0
        anchor 0 0.045 -0.0176
      }
      device [
        RotationalMotor {
          name "right wheel motor"
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.044999999999999984 -0.0176
        rotation 2.5443400409160625e-09 -2.544330695050404e-09 -1 1.5708
        children [
          Transform {
            rotation -0.5771969549958951 0.5776567755050862 -0.5771969549958951 2.0939354039397986
            children [
              Shape {
                appearance PBRAppearance {
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject Transform {
          rotation -0.5771969549958951 0.5776567755050862 -0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.02
              }
            }
          ]
        }
        physics USE WHEEL_PHYSICS
      }
    }
    DEF YELLOW_ROBOT_SHAPE Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "soccer/yellow.png"
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
    Solid {
      translation 0 0.0385 0.02
      rotation 0 0 1 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
        DistanceSensor {
          name "distancesensor left"
          lookupTable [
            0 0 0
            1 1000 0.05
          ]
          type "sonar"
        }
      ]
      name "left sonar"
    }
    Solid {
      translation 0 -0.0385 0.02
      rotation 0 0 1 -1.5707996938995747
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
        DistanceSensor {
          name "distancesensor right"
          lookupTable [
            0 0 0
            1 1000 0.05
          ]
          type "sonar"
        }
      ]
      name "right sonar"
    }
    Solid {
      translation 0.0385 0 0.02
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
        DistanceSensor {
          name "distancesensor front"
          lookupTable [
            0 0 0
            1 1000 0.05
          ]
          type "sonar"
        }
      ]
      name "front sonar"
    }
    Solid {
      translation -0.0385 0 0.02
      rotation 0 0 1 3.141592653589793
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
        DistanceSensor {
          name "distancesensor back"
          lookupTable [
            0 0 0
            1 1000 0.05
          ]
          type "sonar"
        }
      ]
      name "back sonar"
    }
    Receiver {
      name "supervisor receiver"
      channel 1
      allowedChannels [
        1
      ]
    }
    Receiver {
      name "team receiver"
      channel 3
      allowedChannels [
        3
      ]
    }
    Receiver {
      name "ball receiver"
      type "infra-red"
      channel 5
      allowedChannels [
        5
      ]
    }
    Emitter {
      name "team emitter"
      range 10
      channel 3
      baudRate 115200
      allowedChannels [
        3
      ]
    }
    GPS {
    }
    Compass {
    }
  ]
  name "Y2"
  boundingObject USE YELLOW_ROBOT_SHAPE
  physics USE ROBOT_PHYSICS
  controller "robots"
  synchronization FALSE
}
Robot {
  children [
    Emitter {
      range 10
      channel 1
      baudRate 115200
      allowedChannels [
        1
      ]
    }
  ]
  controller "rcj_soccer_referee_supervisor"
  supervisor TRUE
  synchronization TRUE
}
